proto/v1/asset.proto

```diff
@@ -0,0 +1,90 @@
+  // The amount of the asset.
+  // In cases of AssetType being used as the to field, it doesn't have to provided
+  // and can be left empty
+  string amount = 3 [(buf.validate.field).string = {pattern: "^[0-9]+$",}, (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED];  
+  string chain_id = 4 [(buf.validate.field).string = {min_len: 1, pattern: "^[1-9][0-9]*$"}];  // The chain ID where the asset resides.
+}
+
+// Message representing the details of a stake.
+message StakeType {
+  AssetKind type = 1 [(buf.validate.field).enum = {defined_only: true}];  // The type of the stake.
+  string address = 2 [(buf.validate.field).string = {pattern: "^0x[a-fA-F0-9]{40}$"}];  // The address of the stake.
+  string amount = 3 [(buf.validate.field).string = {pattern: "^[0-9]+$"}];  // The amount of the stake.
+  string chain_id = 4 [(buf.validate.field).string = {min_len: 1, pattern: "^[1-9][0-9]*$"}];  // The chain ID where the asset resides.
+}
+
+// Message representing the details of a loan.
+message LoanType {
+  AssetKind type = 1 [(buf.validate.field).enum = {defined_only: true}];  // The type of the loan.
+  string asset = 2 [(buf.validate.field).string = {len:42, pattern: "^0x[a-fA-F0-9]{40}$"}];  // The asset associated with the loan.
+  string amount = 3 [(buf.validate.field).string = {pattern: "^[0-9]+$"}];  // The amount of the loan.
+  string address = 4 [(buf.validate.field).string = {len:42, pattern: "^0x[a-fA-F0-9]{40}$"}];  // The address associated with the loan.
+  string chain_id = 5 [(buf.validate.field).string = {min_len: 1, pattern: "^[1-9][0-9]*$"}];  // The chain ID where the asset resides.
+}
+
+// Message representing additional data for an intent.
+message ExtraData {
+  google.protobuf.BoolValue partially_fillable = 1;  // Indicates if the intent is partially fillable.
+}
+
+// Message representing an intent with various types of transactions.
+message Intent {
+  string sender = 1 [(buf.validate.field).string = {len:42, pattern: "^0x[a-fA-F0-9]{40}$"}];  // The sender of the intent.
+  
+  // Oneof field representing the asset being sent.
+  oneof from {
+    AssetType from_asset = 2;  // The asset being sent.
+    StakeType from_stake = 3;  // The stake being sent.
+    LoanType from_loan = 4;    // The loan being sent.
+  }
+
+  // Oneof field representing the asset being received.
+  oneof to {
+    AssetType to_asset = 5;  // The asset being received.
+    StakeType to_stake = 6;  // The stake being received.
+    LoanType to_loan = 7;    // The loan being received.
+  }
+
+  ExtraData extra_data = 8;  // Additional data for the intent.
+  ProcessingStatus status = 9;  // The processing status of the intent.
+  int64 created_at = 10 [(buf.validate.field).int64 = {gt: 0}];  // The creation timestamp of the intent.
+  int64 expiration_at = 11 [(buf.validate.field).int64 = {gte: 0}];  // The expiration timestamp of the intent.
```

Comment:
use `google.protobuf.timestamp`

Reply:
On that already. Broke a few tests. Fixing